lib/modules/datasource/custom/index.ts

```diff
@@ -46,13 +46,22 @@ export class CustomDatasource extends Datasource {
     logger.trace({ data }, `Custom manager fetcher '${format}' returned data.`);
 
     for (const transformTemplate of transformTemplates) {
+      const expression = jsonata.getExpression(transformTemplate);
+
+      if (expression instanceof Error) {
+        logger.once.warn(
+          { errorMessage: expression.message },
+          `Error while compiling JSONata expression: ${transformTemplate}`,
+        );
+        return null;
+      }
+
       try {
-        const expression = jsonata(transformTemplate);
         data = await expression.evaluate(data);
       } catch (err) {
```

Comment:
this should be checked in utils to not repeat the pattern

Reply:
What's the benefit if it's one line?

Reply:
never mind, it's now gone so this comment is obsolete

Reply:
Actually I brought it back

Reply:
I made your suggested change but didn't like it, prefer this way unless I'm missing something